apps/sales/services.py

```
@@ -0,0 +1,82 @@
+        
+        # 1. Prepare basics
+        total_amount = Decimal('0.00')
+        
+        # Calculate expected total first to ensure data integrity
+        for item in items_data:
+            product = item['product']
+            quantity = item['quantity']
+            
+            # Stock Check
+            if product.current_stock < quantity:
+                raise ValidationError(f"Stock insuffisant pour {product.name}. Stock: {product.current_stock}")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)